### PR TITLE
fix(ui): added check to ensure that windowperiod isn't calculated when the query is undefined

### PR DESCRIPTION
**Pull request by Asalem1**
Closes https://app.honeybadger.io/projects/61122/faults/58676136

### Problem

The `FunctionSelector` component was triggering the `getWindowPeriod` function with an empty query when the component was initialized. The query would be blank at this time, so the `getMinDurationFromAST` function would throw an error, resulting a repetitive (not blocking) honeybadger error

### Solution

Added a check before running the `getWindowPeriod` function to see if the `query` is undefined or an empty string to ensure that the error won't be thrown due to an `invalid time range`

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)

### CODE REVIEW

File: ui/src/variables/utils/getWindowVars.ts [hunk 1]
```diff
@@ -51,6 +51,9 @@ export const getWindowPeriod = (
   query: string,
   variables: VariableAssignment[]
 ): number | null => {
+  if (query === undefined || query.length === 0) {
```
Comment (ebb-tide) [inline ui/src/variables/utils/getWindowVars.ts]: It makes me nervous that I don't know if and when query can be undefined here. 
I think we should only return null if `(!query.length)`

File: ui/src/variables/utils/getWindowVars.ts [hunk 2]
```diff
@@ -64,7 +67,8 @@ export const getWindowPeriod = (
 
     return Math.round(queryDuration / DESIRED_POINTS_PER_GRAPH)
   } catch (error) {
-    reportError(error)
+    console.error(error)
+    reportError(error, {component: 'getWindowPeriod function'})
```
Comment (drdelambre) [inline ui/src/variables/utils/getWindowVars.ts]: can we add the query here too?

Comment (Asalem1) [reply ui/src/variables/utils/getWindowVars.ts]: We should be able to add that

Comment (hoorayimhelping) [inline ui/src/variables/utils/getWindowVars.ts]:
```suggestion
    reportError(error, {context: { query }, component: 'getWindowPeriod function'})
```

File: ui/src/shared/utils/errors.ts [hunk 3]
```diff
@@ -12,6 +12,8 @@ if (CLOUD) {
 
 interface AdditionalOptions {
   component?: string
+  context?: object
```
Comment (hoorayimhelping) [inline ui/src/shared/utils/errors.ts]: any reason this shouldn't be

```ts
context?: {[key: string]: any}
```
?

it's a little more descriptive and makes it easier to limit the values allowed if we find that `any` is too open.

**Pull request merged**